kura/org.eclipse.kura.driver.opcua.provider/src/main/java/org/eclipse/kura/internal/driver/opcua/OpcUaDriver.java

```diff
@@ -350,72 +310,70 @@ public synchronized ChannelDescriptor getChannelDescriptor() {
         }
     }
 
+    private Object extractValue(DataValue value) throws IllegalArgumentException {
+        Variant variant = value.getValue();
+        if (variant == null) {
+            throw new NoSuchElementException(message.errorNullResult());
```

Comment:
Since in most of the source base in Kura, null parameters throw NPE, I believe it is better to switch to NPE.